### PR TITLE
refactor: [M3-5181] - RQ-ify Events

### PR DESCRIPTION
**Pull request by hkhalil-akamai**
## Description 📝
Part 1: [NotificationMenu, EventsLanding (#9046)](https://github.com/linode/manager/pull/9046)
Part 2: [Event Handlers (#9293)](https://github.com/linode/manager/pull/9293)

Migrates the Redux Events store to RQ with the new `useEventsInfiniteQuery` hook.

## What exactly do Events do ❓ *(Copied from Part 2)*
Events have a few different functions in Cloud Manager:

* Notify the user of ongoing/completed changes and updates to their account entities (Linodes, NodeBalancers, etc.)
    * These Event notifications are displayed in the Notification Menu and the Events Landing at /events.
    * Toast notifications appear when long running tasks complete (i.e., a Linode migration)
* Trigger event handlers that update locally cached data
    * E.g., when a Linode is updated in another tab, the Linodes cache is invalidated and Cloud Manager requests and displays the latest details without having to refresh the page
* Display the current status of an entity
    * E.g., when a Linode resize is in progress, this information, along with the progress will be displayed in the Linode details page

## Major Changes 🔄
- New `useEventsInfiniteQuery` hook to handle fetching events and registering event callback handlers
- New `withEventsInfiniteQuery` HOC to enable class-based components to use events
- Refactor `EventsLanding` and `NotificationMenu` components to fetch and display events using the new hook
- Consolidate events handlers in `useAppEventHandlers` and `useToastNotifications` hooks in `App.tsx`
- Remove `rxjs` as a dependency
- Delete the Events Redux store and all associated files (reducers, actions, helpers, etc.)

## How to test 🧪
1. Follow the [testing steps in Part 1](https://github.com/linode/manager/pull/9046#:~:text=How%20to%20test,%F0%9F%A7%AA) to ensure **`EventsLanding` and `NotificationMenu`** work as expected and there are no regressions from prod.
2. Follow the [testing steps in Part 2](https://github.com/linode/manager/pull/9293#:~:text=How%20to%20test,%F0%9F%A7%AA) to ensure **event handlers and toasts** work as expected and there are no regressions from prod.
3. Verify all unit tests and E2E tests pass.

### CODE REVIEW

File: packages/manager/src/features/Events/EventsLanding.tsx [hunk 1]
```diff
@@ -392,20 +211,14 @@ export const renderTableBody = (
...
   entitiesLoading: areEntitiesLoading(state.__resources),
-  eventsFromRedux: state.events.events,
-  inProgressEvents: state.events.inProgressEvents,
-  mostRecentEventTime: state.events.mostRecentEventTime,
 });
 
 const connected = connect(mapStateToProps);
 
-const enhanced = compose<CombinedProps, Props>(connected);
+const enhanced = compose<CombinedProps, EventsLandingProps>(connected);
```
Comment (bnussman-akamai) [inline packages/manager/src/features/Events/EventsLanding.tsx:222]: These lines don't need to exist. You can remove them now or when you remove Linodes from Redux completely.

File: packages/manager/src/features/TopMenu/NotificationMenu/NotificationMenu.styles.tsx
Comment (bnussman-akamai) [inline packages/manager/src/features/TopMenu/NotificationMenu/NotificationMenu.styles.tsx:1]: If this file has no tsx is can be `.ts`

**Review by mjac0bs — APPROVED** [review]
Thanks for all the work you put into this!

I'm not seeing any regressions after testing various events and the various event landing, activity feed, and notifications menu views.

**Review by bnussman-akamai — APPROVED** [review]
Amazing job taking this task on 🎉 I didn't find any bugs or unexpected behavior during my testing today. 

Will keep a close eye on it when it's in develop but I'm feeling good!

File: packages/manager/src/features/Linodes/LinodesDetail/LinodeResize/LinodeResize.tsx [hunk 2]
```diff
@@ -1,5 +1,6 @@
 import { Disk, LinodeType } from '@linode/api-v4/lib/linodes';
 import { APIError } from '@linode/api-v4/lib/types';
+import { Typography } from '@mui/material';
```
[the remark this answers is not in the document]
Comment (hkhalil-akamai) [reply packages/manager/src/features/Linodes/LinodesDetail/LinodeResize/LinodeResize.tsx:3]: No, I just wasn't paying attention!

File: packages/manager/src/features/TopMenu/NotificationMenu/NotificationMenu.tsx [hunk 3]
```diff
@@ -54,11 +45,13 @@ const NotificationIconBadge = styled('div')(({ theme }) => ({
...
   const theme = useTheme();
 
+  const notificationContext = React.useContext(_notificationContext);
+
   const { dismissNotifications } = useDismissibleNotifications();
   const { data: notifications } = useNotificationsQuery();
   const formattedNotifications = useFormattedNotifications();
   const eventNotifications = useEventNotifications();
-  const notificationContext = React.useContext(_notificationContext);
+  const { mutateAsync: markEventsAsSeen } = useMarkEventsAsSeen();
```
[the remark this answers is not in the document]
Comment (hkhalil-akamai) [reply packages/manager/src/features/TopMenu/NotificationMenu/NotificationMenu.tsx:54]: So there's a bit of history that's worth mentioning here. In Part 1, I experimented with using both `seen` and `read`: `seen` for events that the user has already seen (i.e., they opened the NotificationMenu) and `read` for events that have been received by the Cloud Manager (to know which events we already dispatched to event handlers).

Ultimately, there were a few flaws to that approach, such as not supporting multiple concurrent clients as well increased API traffic (since every event generated an API call). So we reverted back to the old system of keeping track of the last event's timestamp and only requesting events after that timestamp.

tl;dr: `read` was used at one incarnation of this refactor, but is not anymore. However, thanks to Banks's work, it can now be filtered on so it can be used if needed in the future.

[the remark this answers is not in the document]
Comment (mjac0bs) [reply packages/manager/src/features/TopMenu/NotificationMenu/NotificationMenu.tsx:54]: Thank you! I'm glad to have this context as we bid you farewell for a bit. :)
